### DISCUSSION TITLE
Access registered refs in nested form to manually focus fields that have errors set with setError post submit.

### DISCUSSION THREAD

**Discussion by GV94** [react-hook-form/react-hook-form#2666] [Q&A]
What the title says,

I have an API that handles the submitted form and does server side validation. If the validation fails server side an error object is returned to the form, I use setError to apply the errors to the fields. However I need to focus the first field that has an error, but as I am using nested fields and useFormContext, I don't know how to access the refs of these fields.

I hope my question is clear, if it isn't don't hesitate to ask.

Thanks in advance!

**Comment by bluebill1049** [comment]
you can use `errors.test.ref.focus()` that sepcific error.
